tests/validation/tests/v3_api/test_app.py

```diff
@@ -105,246 +119,725 @@ def test_tiller():
+        project_detail["cluster2"]["namespace1"] = \
+        create_project_and_ns(USER_TOKEN,
+                              cluster2,
                               random_test_name("testapp"))
-    project_detail["cluster"] = clusters[0]
-    #create users
-    user_token["user_c1_p1_owner"]["user"], \
-        user_token["user_c1_p1_owner"]["token"] = create_user(client)
-    user_token["user_c1_p1_member"]["user"], \
-        user_token["user_c1_p1_member"]["token"] = create_user(client)
+    project_detail["cluster1"]["cluster"] = cluster1
+    project_detail["cluster2"]["cluster"] = cluster2
+
+    catalog = admin_client.create_catalog(
+        name=CATALOG_NAME,
+        baseType="catalog",
+        branch=BRANCH,
+        kind="helm",
+        url=CATALOG_URL)
+    time.sleep(5)
+
+    # create users
     user_token["user_c1_p2_owner"]["user"], \
-        user_token["user_c1_p2_owner"]["token"] = create_user(client)
+        user_token["user_c1_p2_owner"]["token"] = create_user(admin_client)
+    user_token["user_c2_p1_owner"]["user"], \
+        user_token["user_c2_p1_owner"]["token"] = create_user(admin_client)
+    user_token["user_c2_owner"]["user"], \
+        user_token["user_c2_owner"]["token"] = create_user(admin_client)
     user_token["user_standard"]["user"], \
-        user_token["user_standard"]["token"] = create_user(client)
-    #Assign roles to the users
-    assign_members_to_project(client,
-                              user_token["user_c1_p1_owner"]["user"],
-                              project_detail["project"],
+        user_token["user_standard"]["token"] = create_user(admin_client)
+
+    # Assign roles to the users
+    assign_members_to_project(admin_client,
+                              user_token["user_c1_p2_owner"]["user"],
+                              project_detail["cluster1"]["project2"],
+                              "project-owner")
+    assign_members_to_project(admin_client,
+                              user_token["user_c2_p1_owner"]["user"],
+                              project_detail["cluster2"]["project1"],
                               "project-owner")
-    assign_members_to_project(client,
-                              user_token["user_c1_p1_member"]["user"],
-                              project_detail["project"],
-                              "project-member")
+    assign_members_to_cluster(admin_client,
+                              user_token["user_c2_owner"]["user"],
+                              project_detail["cluster2"]["cluster"],
+                              "cluster-owner")
+
 
     def fin():
-        client = get_user_client()
-        client.delete(project_detail["project"])
+        admin_client.delete(project_detail["cluster1"]["project2"])
+        admin_client.delete(project_detail["cluster2"]["project1"])
+        admin_client.delete(catalog)
+        for userkey in user_token:
+            admin_client.delete(user_token[userkey]["user"])
+
```

Comment:
How did you avoid deleting users from `rbac_data` from common.py?

Reply:
I think you are right. I will change this code to delete only what is being created in test-app.py